media/js/firefox/desktop/tips.js

```diff
@@ -15,8 +15,12 @@
 
     // only show download button for users on desktop platforms, using either a non-Firefox browser
     // or an out of date version of Firefox
-    if ((window.isFirefox() && window.isFirefoxUpToDate()) || $html.hasClass('android') || $html.hasClass('ios') || $html.hasClass('fxos')) {
-        $('#footer').addClass('hide-download');
+    if (window.isFirefox() || $html.hasClass('android') || $html.hasClass('ios') || $html.hasClass('fxos')) {
```

Comment:
We need to rethink conditional logic such as this. `getFirefoxDetails()` should only be called for Firefox Desktop users. Other browsers / platforms should not be penalized with delayed loading times, especially for mobile.